engine-precompiles/src/alt_bn256.rs

```diff
@@ -42,23 +42,115 @@ mod consts {
 
     /// Pair element length.
     pub(super) const PAIR_ELEMENT_LEN: usize = 192;
+
+    pub(super) const SCALAR_PART_LEN: usize = SCALAR_LEN / 2;
+
+    /// Size of BN scalars.
+    pub(super) const SCALAR_LEN: usize = 32;
+
+    /// Half the size of a point size.
+    pub(super) const POINT_PART_LEN: usize = POINT_LEN / 2;
+
+    /// Size of BN points.
+    pub(super) const POINT_LEN: usize = 64;
+
+    /// Size of BN pairs.
+    pub(super) const POINT_PAIR_LEN: usize = 128;
+
+    /// Output length.
+    pub(super) const OUTPUT_LEN: usize = 64;
+}
+
+#[cfg(feature = "contract")]
+trait HostFnEncode {
+    type Encoded;
+
+    fn sdk_encode(self) -> Self::Encoded;
```

Comment:
Could rename to `host_fn_encode` to match the trait.

Reply:
missed!